pkg/controllers/disruption/consolidation.go

```diff
@@ -192,6 +194,74 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 		results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
 	}
 
+	results.NewNodeClaims[0].NodeClaimTemplate.InstanceTypeOptions = instanceTypesOptionsFilteredByPrice
+	return Command{
+		candidates:   candidates,
+		replacements: results.NewNodeClaims,
+	}, nil
+}
+
+// Compute command to execute spot-to-spot consolidation if:
+//  1. The SpotToSpotConsolidation feature flag is set to true.
+//  2. For single-node consolidation:
+//     a. There are at least 15 cheapest instance type replacement options to consolidate.
+//     b. The current candidate is NOT part of the first 15 cheapest instance types inorder to avoid repeated consolidation.
+func (c *consolidation) computeSpotToSpotConsolidation(ctx context.Context, candidates []*Candidate, results *pscheduling.Results,
+	candidatePrice float64) (Command, error) {
+
+	// Spot consolidation is turned off.
+	if !options.FromContext(ctx).FeatureGates.SpotToSpotConsolidation {
+		if len(candidates) == 1 {
+			c.recorder.Publish(disruptionevents.Unconsolidatable(candidates[0].Node, candidates[0].NodeClaim, "SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node")...)
+		}
+		logging.FromContext(ctx).Debugf("SpotToSpotConsolidation is disabled, can't replace a spot node with a spot node.")
```

Comment:
Yeah, it can turn out to be spammy if there are huge number of spot instances. I tested with fewer spot instances and the disruption loop was kicking in every 5 mins. So, it spams every 5 mins or whenever there is a cluster state change.

I think it would be right move to actually remove this. Will remove this in the next iteration.

Reply:
The 5 minutes is only in the case where nothing happened on the previous loop, meaning that cluster state was consolidated. That means this is the least amount of logs we'd see. If we have 100 spot nodes, but other on-demand nodes can be consolidated, we'll still get these logs when in reality the on-demand ones are the only nodes we can delete. 

Even though the default is info, i'm still hesitant to add this in, as the more you have logs spam, the less easy it is to parse through everything. Really, I think what would be best is if we could just do one log every 5 minutes to alert the user, but not spam them, but I'm not sure how that'd be implemented.

Reply:
> Generic question : If we had to debug in future, do we get to know if a customer has set the flag as true or false if we remove this log ?

Good question, the only way to know if the flag is set is if you get event/log. You could also look at the environment variables in the deployment if you have access to it.

Reply:
I have removed this log line.